llama-index-core/llama_index/core/indices/prompt_helper.py

```diff
@@ -214,6 +229,15 @@ def _get_available_chunk_size(
             prompt_str = get_empty_prompt_txt(prompt)
             num_prompt_tokens = self._token_counter.get_string_tokens(prompt_str)
 
+        num_prompt_tokens += self._token_counter.estimate_tokens_in_tools(
+            [x.metadata.to_openai_tool() for x in tools]
+        )
+
+        if llm is not None:
+            num_prompt_tokens += self._token_counter.get_string_tokens(
+                llm.system_prompt or ""
```

Comment:
this is a bit hacky due to the implementation of StructuredLLM, but if StructuredLLM is used it cannot and should not have a `system_prompt` set, even though the underlying LLM it wraps does have a `system_prompt`.

You might have to do a special check for `llm.llm.system_prompt` if type StructuredLLM here

Reply:
fixed